packages/ra-ui-materialui/src/form/SimpleFormIterator.js

```diff
@@ -65,7 +65,11 @@ export class SimpleFormIterator extends Component {
         // but redux-form doesn't provide one (cf https://github.com/erikras/redux-form/issues/2735)
         // so we keep an internal map between the field position and an autoincrement id
         this.nextId = 0;
-        this.ids = props.fields ? props.fields.map(() => this.nextId++) : [];
+        this.ids = props.defaultValue
+            ? props.defaultValue.map(() => this.nextId++)
```

Comment:
this one I don't understand. Can you explain the rationale?

Reply:
Explained in comments